testsuite/tests/typing-layouts/basics_alpha.ml

```diff
@@ -636,15 +636,16 @@ module M9_4 = struct
     | ({vur_void = _},i) -> i
 end;;
 [%%expect {|
-Line 4, characters 8-16:
+Line 4, characters 7-21:
 4 |     | ({vur_void = _},i) -> i
-            ^^^^^^^^
-Error: The record field "vur_void" belongs to the type "void_unboxed_record"
-       but is mixed here with fields of type "('a : value)"
+           ^^^^^^^^^^^^^^
+Error: This pattern matches values of type "void_unboxed_record"
+       but a pattern was expected which matches values of type
+         "('a : value_or_null)"
        The layout of void_unboxed_record is void
          because of the definition of void_unboxed_record at line 12, characters 0-60.
        But the layout of void_unboxed_record must be a sublayout of value
-         because it's a boxed record type.
+         because it's the type of a tuple element.
```

Comment:
I'm not sure...  I'll spend some time looking in case this is a symptom of an actual problem.

Reply:
Aha - the culprit is that we *no longer* assume that all `Maybe_a_record_type` record patterns are value. [Minimal working example on `main`.](https://github.com/ocaml-flambda/flambda-backend/commit/c903249d2297c9352e14123a8a690213d36e6288)

This assumption caused the "boxed records must be value" error to fire before the "tuple elements must be value" error had a chance to.

Our change should be useful anyway for when we allow non-value `[@@unboxed]` records. I don't think it can be harmful, because the sort variable is unified with the record labels' `lbl_res` anyway in `solve_Ppat_record_field`.